extensions-contrib/thrift-extensions/src/main/java/io/druid/data/input/thrift/ThriftInputRowParser.java

```diff
@@ -0,0 +1,169 @@
+  final private TSerializer serializer = new TSerializer(new TSimpleJSONProtocol.Factory());
+  final private Parser<String, Object> parser;
+  volatile private Class<TBase> thriftClass = null;
+
+  @JsonCreator
+  public ThriftInputRowParser(
+      @JsonProperty("parseSpec") ParseSpec parseSpec,
+      @JsonProperty("thriftJar") String jarPath,
+      @JsonProperty("thriftClass") String thriftClassName,
+      @JsonProperty("protocol") String protocol
+  )
+  {
+    this.protocol = protocol;
+
+    if (protocol == null || protocol.equals("compact")) {
+      deserializer = new TDeserializer(new TCompactProtocol.Factory());
+    } else if (protocol.equals("binary")) {
+      deserializer = new TDeserializer(new TBinaryProtocol.Factory());
+    } else {
+      throw new IAE("Unsupport protocol [%s], please use one of [compact, binary]", protocol);
+    }
+
+    this.jarPath = jarPath;
+    this.thriftClassName = thriftClassName;
+    Preconditions.checkNotNull(thriftClassName, "thrift class name");
+
+    this.parseSpec = parseSpec;
+    this.parser = parseSpec.makeParser();
+  }
+
+  protected Class<TBase> getThriftClass()
+      throws IOException, ClassNotFoundException, IllegalAccessException, InstantiationException
+  {
+    final Class<TBase> thrift;
+    if(jarPath != null){
+      File jar = new File(jarPath);
+      URLClassLoader child = new URLClassLoader(
+          new URL[]{jar.toURI().toURL()},
+          this.getClass().getClassLoader()
+      );
+      thrift = (Class<TBase>) Class.forName(thriftClassName, true, child);
+    } else {
+      thrift = (Class<TBase>) Class.forName(thriftClassName);
+    }
+    thrift.newInstance();
+    return thrift;
+  }
+
+
+  @Override
+  public InputRow parse(Object input)
+  {
+    // There is a Parser check in phase 2 of mapreduce job, thrift jar may not present in peon side.
+    // Place it this initialization in constructor will get ClassNotFoundException
+    try {
+      if(thriftClass == null) {
+        thriftClass = getThriftClass();
+      }
+    }
+    catch (IOException e) {
+      throw new IAE(e, "failed to load jar [%s]", jarPath);
+    }
+    catch (ClassNotFoundException e) {
+      throw new IAE(e, "class [%s] not found in jar", thriftClassName);
+    }
+    catch (InstantiationException e) {
+      throw new IAE(e, "instantiation thrift instance failed");
+    }
+    catch (IllegalAccessException e) {
+      throw new IAE(e, "instantiation thrift instance failed, IllegalAccessException");
+    }
+
+    try {
+      final byte[] bytes;
+      if (input instanceof BytesWritable) {
+        bytes = ((BytesWritable) input).getBytes();
+      } else if (input instanceof ByteBuffer) {
+        bytes = ((ByteBuffer) input).array();
+      } else {
+        throw new IAE("unsupport input class of [%s]", input.getClass());
+      }
+
+      TBase o = thriftClass.newInstance();
+      deserializer.deserialize(o, bytes);
+      String json = serializer.toString(o);
```

Comment:
if you are just converting thrift to Json, you dont need to pass in a parser, just use the Jsonparser?